### PR TITLE
Use Hyperlink's Hypothesis strategies

### PR DESCRIPTION
**Pull request by wsanchez**
Use Hypothesis strategies that are now provided by Hyperlink, remove them from Klein.

These were copied there from Klein to make them more available to projects and manage them with Hyperlink itself.

### CODE REVIEW

**Comment by wsanchez** [comment]
Build is [failing due to a missing file](https://travis-ci.org/github/twisted/klein/jobs/722089054#L575) from Hyperlink.

I've opened a PR for that: python-hyper/hyperlink#134

**Comment by wsanchez** [comment]
OK, hyperlink has a fix on master

File: src/klein/test/test_headers.py [hunk 1]
```diff
@@ -36,6 +54,59 @@
 __all__ = ()
 
 
+T = TypeVar("T")
```
Comment (wsanchez) [inline src/klein/test/test_headers.py:57]: This section moved here; wasn't needed by Hyperlink.

File: tox.ini [hunk 2]
```diff
@@ -33,10 +33,10 @@ deps =
     tw1910: Twisted==19.10.0
     tw203: Twisted==20.3.0
     twcurrent: Twisted
-    twtrunk: https://github.com/twisted/twisted/archive/trunk.zip
+    twtrunk: git+git://github.com/twisted/twisted@trunk#egg=Twisted
 
     attrs==20.1.0
-    hyperlink==20.0.1
```
Comment (wsanchez) [inline tox.ini:39]: Until the next Hypothesis release.

File: tox.ini [hunk 3]
```diff
@@ -33,10 +33,10 @@ deps =
     tw1910: Twisted==19.10.0
     tw203: Twisted==20.3.0
     twcurrent: Twisted
-    twtrunk: https://github.com/twisted/twisted/archive/trunk.zip
+    twtrunk: git+git://github.com/twisted/twisted@trunk#egg=Twisted
```
Comment (twm) [inline tox.ini:30]: Why change this from the archive form?

**Review by twm — APPROVED** [review]
This looks good to me. I had one question on the Tox config, but it will certainly work as-is so I'm approving.

**Pull request merged**